defaults/api_ents_test.go

```diff
@@ -0,0 +1,209 @@
+package defaults
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/moby/libentitlement/entitlement"
+	"github.com/moby/libentitlement/secprofile"
+	"github.com/moby/libentitlement/testutils"
+)
+
+func TestAPIEntitlementEnforceAllow(t *testing.T) {
+	entitlementID := APIEntFullID
+
+	apiEnt, ok := GetDefaultEntitlement(entitlementID)
+	require.True(t, ok, fmt.Errorf("API entitlement not present"))
+
+	ociProfile := secprofile.NewOCIProfile(testutils.TestSpec(), "test-profile")
+
+	testAPIIDStr := "foo"
+	testAPIID := secprofile.APIID(testAPIIDStr)
+	// Generate "foo:all:allow"
+	testAPIValue := fmt.Sprintf("%s:%s:%s", testAPIIDStr, APIFullControl, string(secprofile.Allow))
+
+	apiStrEnt, ok := apiEnt.(*entitlement.StringEntitlement)
+	require.True(t, ok, fmt.Errorf("API entitlement is not a string entitlement"))
+
+	err := apiStrEnt.SetValue(testAPIValue)
+	require.NoError(t, err)
+
+	refAPIRights := make(map[secprofile.APIID]map[secprofile.APISubsetID]secprofile.APIAccess)
+	refAPIRights[testAPIID] = map[secprofile.APISubsetID]secprofile.APIAccess{
+		secprofile.APISubsetID(APIFullControl): secprofile.Allow,
+	}
+
+	_, err = apiEnt.Enforce(ociProfile)
+	require.NoError(t, err)
+
+	// Check equality on API identifiers, API subset identifiers and API access rules
+	require.True(t, reflect.DeepEqual(ociProfile.APIAccessConfig.APIRights, refAPIRights))
+}
+
+type invalidProfile struct{}
+
+func (p *invalidProfile) GetType() secprofile.ProfileType {
+	return secprofile.ProfileType("invalid-profile")
+}
+
+func TestAPIEntitlementEnforceErrors(t *testing.T) {
+	entitlementID := APIEntFullID
+
+	apiEnt, ok := GetDefaultEntitlement(entitlementID)
+	require.True(t, ok, fmt.Errorf("API entitlement not present"))
+
+	ociProfile := secprofile.NewOCIProfile(testutils.TestSpec(), "test-profile")
+
+	testAPIIDStr := "foo"
+	// Generate "foo:all"
+	testAPIValue := fmt.Sprintf("%s:%s", testAPIIDStr, APIFullControl)
+
+	apiStrEnt, ok := apiEnt.(*entitlement.StringEntitlement)
+	require.True(t, ok, fmt.Errorf("API entitlement is not a string entitlement"))
+
+	err := apiStrEnt.SetValue(testAPIValue)
+	require.NoError(t, err)
+
+	_, err = apiEnt.Enforce(ociProfile)
+	require.Equal(t, err, fmt.Errorf("Wrong API subset and access format, should be \"api-id:subset:[allow|deny]\""))
```

Comment:
If we plan to add more errors, it'd probably be better to make a test-slice/map for a cleaner function